packages/auth/src/grant/model.ts

```diff
@@ -61,6 +66,70 @@ export class Grant extends BaseModel {
   public interactNonce?: string // AS-generated nonce for post-interaction hash
 }
 
+interface ToOpenPaymentsInteractiveGrantArgs {
+  authServerUrl: string
+  client: {
+    name: string
+    uri: string
+  }
+  waitTimeSeconds?: number
+}
+
+export function toOpenPaymentsInteractiveGrant(
+  grant: Grant,
+  args: ToOpenPaymentsInteractiveGrantArgs
+): OpenPaymentsInteractiveGrant {
+  if (!isInteractiveGrant(grant)) {
+    throw new Error('Expected interactive grant')
+  }
+
+  const { authServerUrl, client, waitTimeSeconds } = args
+
+  const redirectUri = new URL(
+    authServerUrl + `/interact/${grant.interactId}/${grant.interactNonce}`
+  )
+
+  redirectUri.searchParams.set('clientName', client.name)
+  redirectUri.searchParams.set('clientUri', client.uri)
+
+  return {
+    interact: {
+      redirect: redirectUri.toString(),
+      finish: grant.interactNonce
+    },
+    continue: {
+      access_token: {
+        value: grant.continueToken
+      },
+      uri: `${authServerUrl}/auth/continue/${grant.continueId}`,
+      wait: waitTimeSeconds
+    }
+  }
+}
+
+interface ToOpenPaymentsGrantArgs {
+  authServerUrl: string
+}
+
+export function toOpenPaymentsGrant(
+  grant: Grant,
+  args: ToOpenPaymentsGrantArgs,
+  accessToken: AccessToken,
+  accessItems: Access[]
+): OpenPaymentsGrant {
```

Comment:
I kind of like the approach of having this as a standalone function that isn't a method on the `XModel` itself, but takes in a `Model` instance itself. It felt better to do this because of the amount of different arguments we pass in to construct the resulting OpenPayments object. NABD, but would be curious to hear opinion on this.

I could update the `backend` `toOpenPaymentsX` methods as well to this

Reply:
@wilsonianb i'm kind of leaning toward using this method in `backend` too 😬

Reply:
I agree it makes sense in `auth`
I'm still pondering `backend`.